powertools-e2e-tests/src/test/java/software/amazon/lambda/powertools/testutils/Infrastructure.java

```diff
@@ -0,0 +1,325 @@
+        cfn.deleteStack(DeleteStackRequest.builder().stackName(stackName).build());
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        public long timeoutInSeconds = 30;
+        public String pathToFunction;
+        public String testName;
+        private String stackName;
+        private boolean tracing = false;
+        private JavaRuntime runtime;
+        private Map<String, String> environmentVariables = new HashMap<>();
+        private String idemPotencyTable;
+
+        private Builder() {
+            getJavaRuntime();
+        }
+
+        /**
+         * Retrieve the java runtime to use for the lambda function.
+         */
+        private void getJavaRuntime() {
+            String javaVersion = System.getenv("JAVA_VERSION"); // must be set in GitHub actions
+            if (javaVersion == null) {
+                throw new IllegalArgumentException("JAVA_VERSION is not set");
+            }
+            if (javaVersion.startsWith("8")) {
+                runtime = JavaRuntime.JAVA8AL2;
+            } else if (javaVersion.startsWith("11")) {
+                runtime = JavaRuntime.JAVA11;
+            } else {
+                throw new IllegalArgumentException("Unsupported Java version " + javaVersion);
+            }
+            LOG.debug("Java Version set to {}, using runtime {}", javaVersion, runtime.getRuntime());
+        }
+
+        public Infrastructure build() {
+            Objects.requireNonNull(testName, "testName must not be null");
+
+            String uuid = UUID.randomUUID().toString().replace("-", "").substring(0, 12);
+            stackName = testName + "-" + uuid;
+
+            Objects.requireNonNull(pathToFunction, "pathToFunction must not be null");
+            return new Infrastructure(this);
+        }
+
+        public Builder testName(String testName) {
+            this.testName = testName;
+            return this;
+        }
+
+        public Builder pathToFunction(String pathToFunction) {
+            this.pathToFunction = pathToFunction;
+            return this;
+        }
+
+        public Builder tracing(boolean tracing) {
+            this.tracing = tracing;
+            return this;
+        }
+
+        public Builder idempotencyTable(String tableName) {
+            this.idemPotencyTable = tableName;
+            return this;
+        }
+
+        public Builder environmentVariables(Map<String, String> environmentVariables) {
+            this.environmentVariables = environmentVariables;
+            return this;
+        }
+
+        public Builder timeoutInSeconds(long timeoutInSeconds) {
+            this.timeoutInSeconds = timeoutInSeconds;
+            return this;
+        }
+    }
+
+    private Stack createStackWithLambda() {
+        Stack stack = new Stack(app, stackName);
+        List<String> packagingInstruction = Arrays.asList(
+                "/bin/sh",
```

Comment:
Does the powertools test suite otherwise run on windows?

Reply:
Good question... I don't know!

Reply:
[it does](https://github.com/scottgerring/aws-lambda-powertools-java/actions/runs/4261104885/jobs/7415028753) - tested quickly on windows - do we want to maintain that?